### PR TITLE
will add GET articles endpoint with all of the required properties so…

### PR DESCRIPTION
**Pull request by marklovejoydev**
…rted in order of created_at this will also count the number of comments PSA relevent testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -83,3 +83,35 @@ describe('GET /api/articles/:article_id',() => {
...
+    test('200, will return the array sorted by created_at', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({ body }) => {
+            const { articles } = body
+            expect(articles).toBeSortedBy('created_at', { descending: true })
+        })
+    })
+})
```
Comment (corkesw) [inline __tests__/app.test.js:117]: Great testing, I'm really liked the use if regex to validate the datetime string :)

File: controller/article-controller.js [hunk 2]
```diff
@@ -8,4 +8,12 @@ exports.getArticleById = (req, res, next) => {
...
     
+}
+
+exports.getArticles = (req, res, next) => {
+    selectAllArticles()
+        .then((articles) => {
+            res.status(200).send({articles})
+        })
+        .catch(next)
 }
```
Comment (corkesw) [inline controller/article-controller.js:19]: controller looks good :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -83,3 +83,35 @@ describe('GET /api/articles/:article_id',() => {
...
+            expect(articles).toHaveLength(13)
+            articles.forEach((article) => {
+                expect(article).toHaveProperty('author', expect.any(String));
+                expect(article).toHaveProperty('title', expect.any(String));
+                expect(article).toHaveProperty('article_id', expect.any(Number));
+                expect(article).toHaveProperty('topic', expect.any(String));
+                expect(article).toHaveProperty('created_at', expect.stringMatching(regEx));
+                expect(article).toHaveProperty('votes', expect.any(Number));
+                expect(article).toHaveProperty('article_img_url',expect.any(String));
+                expect(article).toHaveProperty('comment_count', expect.any(String));
```
Comment (corkesw) [inline __tests__/app.test.js:103]: You will have to deal with this being a string in the front end. This is no huge deal but you can nadle it with SQL...

https://www.w3schools.com/sql/func_sqlserver_cast.asp

File: model/article-model.js [hunk 4]
```diff
@@ -10,4 +10,12 @@ exports.selectArticleById = (id) => {
...
     
+}
+
+exports.selectAllArticles = () => {
+let queryString = "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url ORDER BY articles.created_at DESC;"
+return db.query(queryString).then(({ rows }) => {
+  
+  return rows
+})
 }
```
Comment (corkesw) [inline model/article-model.js:21]: Looks good, you cast `cast` the comment count as an integer if you want

**Comment by corkesw** [comment]
Great work, really good tests and solution here. I've given you a few things to consider but there is no need to resubmit before merging.

File: model/article-model.js (hunk 4 above)
Comment (marklovejoydev) [reply model/article-model.js:21]: Thank you thats awesome ive introduced CAST and reflected that in the tests appreciate the direction

**Pull request merged**